pkg/frontend/mysql_protocol.go

```diff
@@ -451,11 +451,6 @@ func (mp *MysqlProtocolImpl) SendPrepareResponse(stmt *PrepareStmt) error {
 			return err
 		}
 	}
-	if numParams > 0 {
```

Comment:
As mentioned in https://dev.mysql.com/doc/dev/mysql-server/latest/page_protocol_com_stmt_prepare.html#sect_protocol_com_stmt_prepare_response,
It needs to satisfy some conditions before not sending EOF.
Add this condition check.